<a href="https://colab.research.google.com/github/SurajLahariya/DS_learning/blob/main/Assignment_3_(_Q_no_3_)_Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

In [2]:
url = 'https://raw.githubusercontent.com/SurajLahariya/DS_learning/main/BuyerRatio.csv'

In [3]:
df = pd.read_csv(url)

In [4]:
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [5]:
Male=pd.Series(df.iloc[0,:])
Male

Observed Values    Males
East                  50
West                 142
North                131
South                 70
Name: 0, dtype: object

In [6]:
Female=pd.Series(df.iloc[1,:])
Female

Observed Values    Females
East                   435
West                  1523
North                 1356
South                  750
Name: 1, dtype: object

In [8]:
df.drop(["Observed Values"], axis =1, inplace=True)

In [9]:
df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [10]:
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [12]:
p_value=stats.chi2_contingency(df)

In [13]:
p_value

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [14]:
p_value[1]

0.6603094907091882

###P-Value (0.6603)> α (0.05), Therefore we can't reject the null Hypothesis.
Proportion of Male and Female across regions is same.